simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -569,61 +789,586 @@ def _eval_tipper_deriv(self, src, mesh, f, du_dm_v=None, v=None, adjoint=False):
+class PointNaturalSource(Impedance):
+    """Point receiver class for magnetotelluric simulations.
+
+    .. warning::
+        This class is deprecated and will be removed in SimPEG v0.24.0.
+        Please use :class:`.natural_source.receivers.Impedance`.
+
+    Assumes that the data locations are standard xyz coordinates;
+    i.e. (x,y,z) is (Easting, Northing, up).
+
+    Parameters
+    ----------
+    locations : (n_loc, n_dim) numpy.ndarray
+        Receiver locations.
+    orientation : {'xx', 'xy', 'yx', 'yy'}
+        MT receiver orientation.
+    component : {'real', 'imag', 'apparent_resistivity', 'phase'}
+        MT data type.
+    """
+
+    def __init__(
+        self,
+        locations=None,
+        orientation="xy",
+        component="real",
+        locations_e=None,
+        locations_h=None,
+        **kwargs,
+    ):
+        if locations is None:
+            if (locations_e is None) ^ (
+                locations_h is None
+            ):  # if only one of them is none
+                raise TypeError(
+                    "Either locations or both locations_e and locations_h must be passed"
+                )
+            if locations_e is None and locations_h is None:
+                warnings.warn(
+                    "Using the default for locations is deprecated behavior. Please explicitly set locations. ",
+                    FutureWarning,
+                    stacklevel=2,
+                )
+                locations_e = np.array([[0.0]])
+                locations_h = locations_e
+        else:  # locations was not None
+            if locations_e is not None or locations_h is not None:
+                raise TypeError(
+                    "Cannot pass both locations and locations_e or locations_h at the same time."
+                )
+            if isinstance(locations, list):
+                if len(locations) == 2:
+                    locations_e = locations[0]
+                    locations_h = locations[1]
+                elif len(locations) == 1:
+                    locations_e = locations[0]
+                    locations_h = locations[0]
+                else:
+                    raise Exception("incorrect size of list, must be length of 1 or 2")
+            else:
+                locations_e = locations_h = locations
+
+        super().__init__(
+            locations_e=locations_e,
+            locations_h=locations_h,
+            orientation=orientation,
+            component=component,
+            **kwargs,
+        )
+
+    def eval(self, src, mesh, f, return_complex=False):  # noqa: A003
+        if return_complex:
+            temp = self.component
+            self.component = "complex"
+            out = super().eval(src, mesh, f)
+            self.component = temp
+        else:
+            out = super().eval(src, mesh, f)
+        return out
+
+    # locations = property(lambda self: self._locations[0], Impedance.locations.fset)
```

Comment:
This line to return the locations that were defined for the soon to be deprecated class breaks a lot of tests. By adding this test, the inherited method which returns the projection matrices doesn't work properly.